src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -24,14 +26,51 @@ private int[] SortedMap(Buffer<TElement> buffer, int minIdx, int maxIdx)
         public IEnumerator<TElement> GetEnumerator()
         {
             Buffer<TElement> buffer = new Buffer<TElement>(_source);
-            if (buffer._count > 0)
+            if (buffer._count >= 32)
             {
-                int[] map = SortedMap(buffer);
+                return GetLargeEnumerator(buffer);
+            }
+            else if (buffer._count > 0)
+            {
+                return GetSmallEnumerator(buffer);
+            }
+
+            return GetEmptyEnumerator();
```

Comment:
If we're going to take this approach, then just letting `GetSmallEnumerator` do its thing with a count of 0 is probably fine.